internal/pubsub/drivertest/drivertest.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package drivertest provides a conformance test for implementations of
+// driver.
+package drivertest
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/google/go-cloud/internal/pubsub"
+)
+
+// Harness descibes the functionality test harnesses must provide to run
+// conformance tests.
+type Harness interface {
+	// MakeTopicDriver creates a Topic and associated Subscription to test.
+	MakePair() (*pubsub.Topic, *pubsub.Subscription, error)
```

Comment:
Do you expect any tests that create more than one pair? For example, a test that creates two subscribers for the same topic.

It might be better to split these (`MakeTopic`, `MakeSubscription`), and provide a way for the test to describe which topic they want. For example, we might require that every implementation provision two topics each with two subscriptions, or whatever, and then these functions would need to describe which one it wants.

Reply:
(this could be a TODO or issue for a separate PR)

Reply:
Filed https://github.com/google/go-cloud/issues/717.